gammapy/time/plot.py

```diff
@@ -30,3 +28,140 @@ def plot_time_difference_distribution(time, ax=None):
+    import math
+
+    ax = plt.gca() if ax is None else ax
+
+    fermi_met_base = astropy.time.Time('2001-01-01T00:00:00')
+
+    fermi_met_start = (time_start - fermi_met_base).sec
+
+    fermi_met_end = (time_end - fermi_met_base).sec
+
+    # As far as I know light curves were only included in 3FGL, so we must use this catalog.
+    fermi_cat = fetch_fermi_catalog('3FGL')
+
+    catalog_index = np.where(fermi_cat[1].data['Source_Name'] == name_3FGL)[0][0]
+
+    time_index_start = np.where(fermi_cat[3].data['Hist_Start'] >= fermi_met_start)[0][0]
+
+    # The final entry is the end of the last bin, so no off by one error
+    time_index_end = np.where(fermi_cat[3].data['Hist_Start'] <= fermi_met_end)[0][-1] + 1
+
+    time_start = fermi_cat[3].data['Hist_Start'][time_index_start: time_index_end]
+    time_end = np.roll(time_start, -1)
+
+    time_diff = time_end - time_start
+
+    # Trim because there is one more bin edge than there is bin mid point
+    time_diff = time_diff[0:-1]
+
+    # Midpoints of each bin.
+    time_mid = time_start[0:-1] + time_diff
+
+    flux_history = fermi_cat[1].data[catalog_index]['Flux_History'][time_index_start: time_index_end]
+
+    flux_history_lower_bound = fermi_cat[1].data[catalog_index]['Unc_Flux_History'][time_index_start: time_index_end, 0]
+    flux_history_upper_bound = fermi_cat[1].data[catalog_index]['Unc_Flux_History'][time_index_start: time_index_end, 1]
+    flux_history_lower_bound = abs(flux_history_lower_bound)
+
+    # Change bins with no flux value from nan to zero.
+    for i in range(0, np.size(flux_history_lower_bound)):
+        if math.isnan(flux_history_lower_bound[i]):
+            flux_history_lower_bound[i] = 0
+
+    # Create array of upper limits where no lower bound was recorded.
+    upper_lims_x = np.copy(time_mid)
+
+    upper_lims_y_start = np.copy(flux_history_upper_bound)
+
+    upper_lims_y = np.copy(flux_history)
+
+    idx = np.where(flux_history_lower_bound <= 0)
+
+    upper_lims_x = upper_lims_x[idx]
+
+    upper_lims_y_start = upper_lims_y_start[idx]
+
+    upper_lims_y = upper_lims_y[idx]
+
+    upper_lims_y_end = np.copy(upper_lims_y_start)
+
+    # Scale the size of the arrow to an arbitrary value.
+    upper_lims_y_end *= -0.3
+
+    # Create an array of data points where a lower bound was recorded.
+    idx = np.where((flux_history_lower_bound) > 0)
+
+    time_mid = time_mid[idx]
+
+    time_diff = time_diff[idx]
+
+    flux_history = flux_history[idx]
+
+    flux_history_upper_bound = flux_history_upper_bound[idx]
+
+    flux_history_lower_bound = flux_history_lower_bound[idx]
+
+    time_mid = (fermi_met_base + astropy.time.TimeDelta(time_mid, format='sec')).plot_date
+    upper_lims_x = (fermi_met_base + astropy.time.TimeDelta(upper_lims_x, format='sec')).plot_date
+
+    # Plot data points and upper limits.
+    plt.errorbar(time_mid, flux_history, yerr=(flux_history_lower_bound, flux_history_upper_bound),
+                 #TODO: x-error bars
```

Comment:
Resolve this TODO?
As you like, plot the x-error bars to indicate the time bins or not.